src/component/handlers/edit/editOnPaste.js

```diff
@@ -124,6 +126,14 @@ function editOnPaste(e: SyntheticClipboardEvent): void {
         );
         return;
       }
+    // Safari does not properly store text/html. Use the internalClipboard
+    // if present and equal to what is on the clipboard.
+    } else if (isSafari && internalClipboard &&
```

Comment:
"Safari does not properly store text/html."  - what you mean by properly ? Also in case that it is a Safari bug would it mean that in the future this snippet would become a code smell ? 

Sorry for being a bit picky in here, but with personal experiences working with multiple text editors in the past as soon as we introduce browser hacks/fixes branching inside the code base the code quality will degenerate very quickly